### ISSUE TITLE
NoSuchMethodError: Class 'HandshakeException' has no instance getter 'stackTrace'

### ISSUE THREAD

**Issue by tiagohm** [flutterchina/dio#345]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | flutter                          |      |
| Platform Version               | 1.7.4                              |      |
| Dio Version                    | 2.1.6                                  |      |
| Android Studio / Xcode Version |               |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Run the following http request in flutter test

```dart
import 'package:dio/dio.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {

  test("NoSuchMethodError", () async {
    final dio = Dio();
    final response = await dio.get("https://expired.badssl.com");
    // print(response);
  });
}
```

```
DioError [DioErrorType.DEFAULT]: NoSuchMethodError: Class 'HandshakeException' has no instance getter 'stackTrace'.
Receiver: Instance of 'HandshakeException'
Tried calling: stackTrace
package:dio/src/dio.dart 713:30  Dio._request.<fn>.<fn>
dart:async                       _AsyncAwaitCompleter.completeError
package:dio/src/dio.dart         Dio._makeRequest
===== asynchronous gap ===========================
dart:async                       Future.catchError
package:dio/src/dio.dart 713:10  Dio._request.<fn>
package:dio/src/dio.dart 976:22  Dio._checkIfNeedEnqueue
package:dio/src/dio.dart 694:9   Dio._request
dart:async                       _AsyncAwaitCompleter.start
package:dio/src/dio.dart 662:34  Dio._request
package:dio/src/dio.dart 606:12  Dio.request
dart:async                       _AsyncAwaitCompleter.start
package:dio/src/dio.dart 597:33  Dio.request
package:dio/src/dio.dart 94:12   Dio.get
test/dio_test.dart 8:32          main.<fn>
dart:async                       _AsyncAwaitCompleter.start
test/dio_test.dart 6:31          main.<fn>
```
```

**Comment by wendux** [comment]
please update to the latest version and try again.

**Comment by tiagohm** [comment]
I updated to latest version and it worked. Thanks!
